File: scripts/map_visualizer.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float32
from std_msgs.msg import Float32MultiArray
from visualization_msgs.msg import Marker
from geometry_msgs.msg import Point


def map_callback(float_array):
    map_data = Float32MultiArray(data=float_array)
    points_marker = Marker()
    points_marker.type = Marker.POINTS
    points_marker.id = 1
    points_marker.header.frame_id = "world"
    points_marker.header.stamp = rospy.Time.now()
    points_marker.ns = "maps"
    points_marker.action = Marker.ADD
    points_marker.pose.orientation.w=1.0
    points_marker.scale.x = 0.1
    points_marker.color.g = 1.0
    points_marker.color.a = 1.0

    for i in range(int(len(map_data.data.data)/2)):
        point = Point()
        point.x = float(map_data.data.data[i*2+0])
        point.y = float(map_data.data.data[i*2+1])
        point.z = float(0.0)
        points_marker.points.append(point)
    marker_pub.publish(points_marker)


rospy.init_node("map_visualizer")
marker_pub = rospy.Publisher("map_vis", Marker, queue_size=10)
canin_sub = rospy.Subscriber("map_data", Float32MultiArray, map_callback)
rospy.spin()

File: src/map_visualizer.cpp
#include "ros/ros.h"
#include "std_msgs/Float32.h"
#include "std_msgs/Float32MultiArray.h"
#include "sensor_msgs/Joy.h"
#include "sensor_msgs/JointState.h"
#include <tf/transform_broadcaster.h>
#include <visualization_msgs/Marker.h>
#include "math.h"
#include <sstream>
#include <string>

std_msgs::Float32MultiArray map_data;
void map_callback(const std_msgs::Float32MultiArray& float_array){
    std_msgs::Float32MultiArray tmp(float_array);
    map_data=tmp;
}

int main(int argc, char **argv)
{
	ros::init(argc, argv, "map_visualizer");
	ros::NodeHandle n;
	ros::NodeHandle pn("~");

	//publish
	ros::Publisher marker_pub = n.advertise<visualization_msgs::Marker>("map_vis", 10);
  //subscribe
  ros::Subscriber canin_sub = n.subscribe("map_data", 10, map_callback);


	ros::Rate loop_rate(100);
	while (ros::ok()){
        //線の設定
        visualization_msgs::Marker line_strip;
        line_strip.header.frame_id="world";
        line_strip.header.stamp=ros::Time::now();
        line_strip.ns="maps";
        line_strip.action=visualization_msgs::Marker::ADD;
        line_strip.pose.orientation.w=1.0;
        line_strip.id=1;
        line_strip.type = visualization_msgs::Marker::LINE_STRIP;
        line_strip.scale.x = 0.01;
        line_strip.color.b = 1.0;
        line_strip.color.a = 1.0;
        //ここpにデータを突っ込むと、Rviz上にその点がマップされる使用
        for(int i=0;i<(int)map_data.data.size()/2;i++){
            geometry_msgs::Point p;
            p.x=map_data.data[i*2+0];
            p.y=map_data.data[i*2+1];
            p.z=0.0;
            line_strip.points.push_back(p);
        }
        marker_pub.publish(line_strip);

		ros::spinOnce();
		loop_rate.sleep();
	}
 	return 0;
}
